lib/perl/Genome/Model/CwlPipeline/Command/Requeue.pm

```diff
@@ -34,23 +40,38 @@ EOHELP
 sub execute {
     my $self = shift;
 
+    my $action = $self->action;
+
     my $count = 0;
     BUILD: for my $build ($self->builds) {
 
-        unless ($build->status eq 'Failed') {
-            $self->error_message('Cannot restart build %s with status %s. Only Failed builds may be restarted.', $build->__display_name__, $build->status);
-            next BUILD;
-        }
+        if ($action eq 'restart') {
+            if ($build->status ne 'Failed') {
+                $self->error_message('Cannot restart build %s with status %s. Only Failed builds may be restarted.', $build->__display_name__, $build->status);
+                next BUILD;
+            }
+
+            unless($build->run_by eq Genome::Config::get('builder_run_as_user')) {
+                $self->error_message('Can only requeue builds for the configured builder user (%s). Restart your own builds directly with `genome model cwl-pipeline restart`.', Genome::Config::get('builder_run_as_user'));
+                next BUILD;
+            }
+        } else {
+            unless($build->run_by eq Genome::Config::get('builder_run_as_user')) {
+                $self->error_message('Can only %s builds for the configured builder user (%s). For your own builds, use `genome model build %s` directly.', $action, Genome::Config::get('builder_run_as_user'), $action);
+                next BUILD;
+            }
 
-        unless($build->run_by eq Genome::Config::get('builder_run_as_user')) {
-            $self->error_message('Can only requeue builds for the configured builder user (%s). Restart your own builds directly with `genome model cwl-pipeline restart`.', Genome::Config::get('builder_run_as_user'));
-            next BUILD;
+            $DB::single = 1;
+            unless($build->model->analysis_project->created_by eq Genome::Sys->username or Genome::Sys->current_user_is_admin) {
```

Comment:
which users are currently admins?  prod-builder and mgi-svc-bga-run?

Reply:
Try: `genome sys user list -f user_roles.name=admin`
If you're `mgi-svc-bga-run`, you can stop and abandon builds directly.  But we could make that user an admin, too.